watcher/watcher.go

```diff
@@ -190,6 +186,88 @@ func (w *Watcher) Start() {
 	extensions.GracefulShutdown(l, w.gracefulShutdown.wg, w.gracefulShutdown.timeout)
 }
 
+func (w *Watcher) watchRooms() error {
+	w.RemoveDeadRooms()
+	w.AutoScale()
+	return nil
+}
+
+// ReportRoomsStatuses runs as a block of code inside WithRedisLock
+// inside a timer tick in w.Start()
+func (w *Watcher) ReportRoomsStatuses() error {
+	if !reporters.HasReporters() {
+		return nil
+	}
+
+	var roomCountByStatus *models.RoomsStatusCount
+	err := w.MetricsReporter.WithSegment(models.SegmentGroupBy, func() error {
+		var err error
+		roomCountByStatus, err = models.GetRoomsCountByStatus(w.RedisClient.Client, w.SchedulerName)
+		return err
+	})
+
+	if err != nil {
+		return err
+	}
+
+	type RoomData struct {
+		Status string
+		Gauge  string
+	}
+
+	roomDataSlice := []RoomData{
+		RoomData{
+			models.StatusCreating,
+			fmt.Sprint(int(roomCountByStatus.Creating)),
```

Comment:
roomCountByStatus.Creating already is an int. Do you need to cast it?

Reply:
You're right!